src/app/components/qualification-overview-component/qualification-overview.component.ts

```diff
@@ -123,12 +121,8 @@ export class QualificationOverviewComponent implements AfterContentInit, OnInit,
   loadTable(): void {
     if (this.selectedCurriculum > 0 && this.selectedLifecycleActivity > 0 && this.selectedArchitecturalLayer > 0) {
       this.backendService.getQualificationTable(this.selectedCurriculum, this.selectedArchitecturalLayer, this.selectedLifecycleActivity)
-        .subscribe(table => {
-          this.table = table;
-          return this.countTotalLearningGoals(this.table);
-        }, () => {
-          this.dataStructure = null;
-        });
+        .subscribe(table => this.countTotalLearningGoals(table),
+        () => this.dataStructure = null);
```

Comment:
I am not sure if dataStructure setting to null won't fail the table generation.

I think it is better to set it to an empty array and add furthermore an initialization to the constructor.